refactor(transformer/styled_components): make unreachable branch `unreachable!`

overlookmotel opened the pull request:
This branch is unreachable, so use `unreachable!` for it, rather than `return`.

Despite `unreachable!` producing more code, this may still be more performant, because it tells the compiler which branch is going to be taken. Compiler will generate assembly to make the branch which is taken the fast path. Either way, it's a micro micro optimization.

Mainly I think it's preferable because it's clearer when reading the code.

overlookmotel commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/12172?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#12172** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/12172?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/12172?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#12170** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/12170?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#12169** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/12169?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#12066** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/12066?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>: 1 other dependent PR ([#12155](https://github.com/oxc-project/oxc/pull/12155) <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/12155?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>)
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=12172).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Dunqing reviewed (approved):
I have been away the transformer too long...

overlookmotel committed:
refactor(transformer/styled_components): make unreachable branch `unreachable!` (#12172)

This branch is unreachable, so use `unreachable!` for it, rather than `return`.

Despite `unreachable!` producing more code, this may still be more performant, because it tells the compiler which branch is going to be taken. Compiler will generate assembly to make the branch which is taken the fast path. Either way, it's a micro micro optimization.

Mainly I think it's preferable because it's clearer when reading the code.
